astroquery/mast/cloud.py

```diff
@@ -86,21 +87,22 @@ def get_cloud_uri(self, data_product, include_bucket=True, full_url=False):
 
         s3_client = self.boto3.client('s3')
 
-        paths = fpl.paths(data_product)
-        if paths is None:
+        path = utils.mast_relative_path(data_product["dataURI"])
+        if path is None:
             raise Exception("Unsupported mission {}".format(data_product['obs_collection']))
```

Comment:
Minor question.  Is "Unsupported mission" the reason that a path wasn't found?  Missions whose URI is "http..." return a valid path, but supplying an unknown `mast:` URI for a supported mission does give a None path.  Just wondering if we can be more informative for the user.

Reply:
Hmmmm yeah. This used to be true(ish) because it was going through the local fpl function which only included the missions where data is in S3. Now.... this part of the code shouldn't really be hit ever. I will change it to a malformed query warning, since it should now only happen if a bad mast uri is given.

Reply:
I'm not sure my new error message is really any more informative to the user, but I'm not really sure how to be clearer. If this error happens at this point it will mean that fabric couldn't find the file on our servers, so it's a larger problem than just cloud access.